app/lib/substrate.py

```diff
@@ -101,13 +106,18 @@ def substrate_sudo_unchecked_weight_call(substrate_client, keypair, payload, wai
 
 
 def substrate_batchall_call(substrate_client, keypair, batch_call, wait=True, sudo=False):
-    call = substrate_client.compose_call(
-        call_module='Utility',
-        call_function='batch',
-        call_params={
-            'calls': batch_call
-        }
-    )
+    # If the batch contains only 1 element, don't use batch
+    if len(batch_call) == 1:
```

Comment:
I introduced this because in `teleport_funds` you had put this logic. So I simply moved it to `substrate_batchall_call`. Since we are wrapping transfer/teleport calls with this batch method, I think it's better to remove the batch when there is only 1 operation.